src/crimson/osd/main.cc

```diff
@@ -38,54 +38,55 @@ seastar::logger& logger() {
 }
 
 void usage(const char* prog) {
-  std::cout << "usage: " << prog << "\n"
-            << "  -i <ID>\n";
+  std::cout << "usage: " << prog << std::endl;
   generic_server_usage();
 }
 
-auto partition_args(seastar::app_template& app, char** argv_begin, char** argv_end)
+auto partition_args(int argc, const char *argv[])
 {
-  // collect all options consumed by seastar::app_template
-  auto parsed = bpo::command_line_parser(std::distance(argv_begin, argv_end),
-                                         argv_begin)
-    .options(app.get_options_description())
-    .style(bpo::command_line_style::default_style &
-           ~bpo::command_line_style::allow_guessing)
-    .allow_unregistered().run();
-  auto unknown_args = bpo::collect_unrecognized(parsed.options,
-                                                bpo::include_positional);
-  std::vector<const char*> ceph_args, app_args;
-  // ceph_argparse_early_args() and
-  // seastar::smp::get_options_description() use "-c" for different
-  // options. and ceph wins
-  auto consume_conf_arg = [&](char** argv) {
-    if (std::strcmp(*argv, "-c") == 0) {
-      std::cout << "warn: apply '-c FILE' as ceph option" << std::endl;
-      ceph_args.push_back(*argv++);
-      if (argv != argv_end) {
-        ceph_args.push_back(*argv++);
-      }
-    }
-    return argv;
-  };
-  auto unknown = unknown_args.begin();
-  auto consume_unknown_arg = [&](char** argv) {
-    for (; unknown != unknown_args.end() &&
-           argv != argv_end &&
-           *unknown == *argv; ++argv, ++unknown) {
-      ceph_args.push_back(*argv);
+  std::vector<const char *> app_args = argv_to_vec(argc, argv);
+  assert(app_args.size() > 0);
+
+  std::vector<const char *> ceph_args;
+  const char* program_name = app_args[0];
+  ConfigValues values;
```

Comment:
```suggestion
  ConfigValues config_values;
```

IMHO, the name of `values` is way too general.

Reply:
I prefer to leave this as values since it's what we already use in config.cc / config.h / config_proxy.h / config_proxy.cc.

Reply:
@markhpc hi Mark.

i think, in different context, we need different names. in `config*`, one can safely assume that we are handling configuration stuff, so it's safe (and might be encouraged) to drop this prefix. just like we don't prefix ever identifiers in our source tree with "ceph", because whoever looking at the tree should be aware that, well, this is "Ceph". but if we are working in a different subsystem or in this case, a higher level, it might be better to be more specific. "value" could be a good name if we are working on a container which is supposed to hold a "value". but this does not hold in this case.

Reply:
will create a PR to address this.